### DISCUSSION TITLE
Getting online during a play sets your score as the user "Guest"

### DISCUSSION THREAD

**Discussion by NotGumballer91** [ppy/osu#19584] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/64581009/183012591-4897b4ed-3f3a-4fe9-af3e-487ff5e3fc49.png)

**To reproduce:**
(Make sure _Stay signed in_ is on)
Disconnect internet
Play a map
Reconnect internet
Finish the map

**Intended behavior:** 
The score should show up as Guest as if you never logged in
or
The score should show up with your logged account

**Actual behavior:**
The score shows up the actual user named "Guest"

**Comment by NotGumballer91** [comment]
Sorry, Duplicate #12407

**Reply by peppy** [reply]
I don't think it's a duplicate? The one you're linking to is asking for the ability to change the "guest" username, while you're reporting your game saving plays as "Guest" when your internet connection is in a bad state.

**Reply by NotGumballer91** [reply]
Ok, This was done on my local release anyway. I'll try it on actual release and see if there's any difference.

**Reply by NotGumballer91** [reply]
It does work
![image](https://user-images.githubusercontent.com/64581009/183017711-3e5e4abc-ade1-4cac-b5c1-2bb9fdd72311.png)

![image](https://user-images.githubusercontent.com/64581009/183017794-f0bcf240-9275-4fd7-8807-afcab9b17429.png)

**Reply by peppy** [reply]
Can you confirm that your toolbar shows your username in those screenshots? ie. the game *did* connect once during startup?

**Reply by NotGumballer91** [reply]
I tried it again _just to be sure._ And yes.

![osu_2022-08-05_15-03-40](https://user-images.githubusercontent.com/64581009/183021122-7d45d226-054d-4fcf-a31f-c1a428caf4ef.png)
![osu_2022-08-05_15-03-47](https://user-images.githubusercontent.com/64581009/183021131-b417822d-f0b5-44fb-a21d-4a352caa5099.png)

**Reply by peppy** [reply]
I can't reproduce this.. and I can't see a code path for how this could happen. The score is created with the current user which is displayed at the point of starting gameplay. Unless you started gameplay before any login could happen (feasible?) then what you are showing is impossible.

I'll see if I can fix the case I just mentioned, as it is a valid one.

**Comment by peppy** [comment, marked as answer]
I've aimed to address this via #19662.
